### PR TITLE
Msk serverless cluster

### LINKED ISSUE #902: Request for `aws_msk_serverless_*` resource
### What resource do you need?

Terraform Resource Name: 

There's a couple resources required for MSK serverless operation:
- [x] [aws_msk_serverless_cluster](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/msk_serverless_cluster.html)
- [ ] [aws_msk_cluster_policy](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/msk_cluster_policy)

### What is your use case?

We've started implementing crossplane and have a requirement from the org to support MSK serverless. 

FWIW I started making an attempt at this myself an am happy to continue with it, but I noticed that the auto generated code I got after moving the `aws_msk_serverless_cluster` to `externalname.go` was under the "msk" API vs the existing Kafka API, is there any way to explicitly set that? 

Thanks in advance!

### PR DESCRIPTION
**Pull request by nalbury**
### Description of your changes

Starts to fix #902 by adding a generated resource for MSK serverless clusters.

I have:

- [X] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

Ran `make e2e UPTEST_EXAMPLE_LIST=examples/kafka/serverlesscluster.yaml` and on the e2e test control plane.

I additionally ran `kubectl apply -f examples/kafka/serverlesscluster.yaml` and verified the resulting infra in AWS:

```
└─> k get -f examples/kafka/serverlesscluster.yaml
NAME                                             READY   SYNCED   EXTERNAL-NAME                                                                                  AGE
serverlesscluster.kafka.aws.upbound.io/example   True    True     arn:aws:kafka:us-east-2:945430712069:cluster/Example/df2a5015-b36e-4b83-ae32-63df77132cb9-s3   5m23s

NAME                                  READY   SYNCED   EXTERNAL-NAME          AGE
securitygroup.ec2.aws.upbound.io/sg   True    True     sg-04bad16bcddcdaeb5   5m23s

NAME                                   READY   SYNCED   EXTERNAL-NAME              AGE
subnet.ec2.aws.upbound.io/subnet-az1   True    True     subnet-03a048c1c2d5226f3   5m23s
subnet.ec2.aws.upbound.io/subnet-az2   True    True     subnet-07080a68648601617   5m22s

NAME                         READY   SYNCED   EXTERNAL-NAME           AGE
vpc.ec2.aws.upbound.io/vpc   True    True     vpc-0bf3a09670f044d85   5m22s
```

### CODE REVIEW

**Comment by nalbury-handy** [comment]
Let me know if there's anything I missed here. Wasn't sure if I should run the e2e tests in Github via the comment syntax?

Thanks!

**Comment by turkenf** [comment]
/test-examples="examples/kafka/serverlesscluster.yaml"

**Comment by nalbury** [comment]
resolved merge conflicts

File: examples/kafka/serverlesscluster.yaml [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+          matchLabels:
+            testing.upbound.io/example-name: sg
+        subnetIdSelector:
+          matchLabels:
+            testing.upbound.io/example-name: subnets
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: SecurityGroup
+metadata:
```
Comment (turkenf) [inline examples/kafka/serverlesscluster.yaml:28]:
```suggestion
metadata:
  annotations:
    meta.upbound.io/example-id: kafka/v1beta1/serverlesscluster
```
Please add the example-id.

File: examples/kafka/serverlesscluster.yaml [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    region: us-east-2
+    vpcIdSelector:
+      matchLabels:
+        testing.upbound.io/example-name: vpc
+
+---
+
+apiVersion: ec2.aws.upbound.io/v1beta1
+kind: Subnet
+metadata:
```
Comment (turkenf) [inline examples/kafka/serverlesscluster.yaml:60]:
```suggestion
metadata:
  annotations:
    meta.upbound.io/example-id: kafka/v1beta1/serverlesscluster
```

**Review by turkenf — COMMENTED** [review]
Thank you for your contribution @nalbury, I left a few commits please check.

**Comment by nalbury** [comment]
@turkenf all requested changes should be in place. Thanks!

**Comment by turkenf** [comment]
/test-examples="examples/kafka/serverlesscluster.yaml"

**Comment by nalbury** [comment]
Noticed several jobs failed with the error:
```
17:10:16 [ .. ] verify go modules dependencies are tidy
go: finding module for package github.com/upbound/upjet/pkg/controller
go: finding module for package github.com/upbound/upjet/pkg/controller/handler
go: downloading github.com/upbound/upjet v0.10.0
go: finding module for package github.com/upbound/upjet/pkg/terraform
go: found github.com/upbound/upjet/pkg/controller in github.com/upbound/upjet v0.10.0
go: found github.com/upbound/upjet/pkg/controller/handler in github.com/upbound/upjet v0.10.0
go: found github.com/upbound/upjet/pkg/terraform in github.com/upbound/upjet v0.10.0
go.mod is not tidy. Please run 'make go.modules.tidy' and stage the changes
17:10:18 [FAIL]
make: *** [build/makelib/golang.mk:221: go.modules.tidy.check] Error 1
```
but in my dev environment:
```
@nalbury ➜ /workspaces/provider-aws (msk-serverless-cluster) $ make submodules 
Submodule 'build' (https://github.com/upbound/build) registered for path 'build'
Cloning into '/workspaces/provider-aws/build'...
Submodule path 'build': checked out '29c0dfa4aff32880e65a3223cccb14f51b39eb2c'
@nalbury ➜ /workspaces/provider-aws (msk-serverless-cluster) $ make go.modules.tidy
18:10:15 [ .. ] mod tidy
18:10:15 [ OK ] mod tidy
@nalbury ➜ /workspaces/provider-aws (msk-serverless-cluster) $ git status
On branch msk-serverless-cluster
Your branch is up to date with 'origin/msk-serverless-cluster'.

nothing to commit, working tree clean
@nalbury ➜ /workspaces/provider-aws (msk-serverless-cluster) $ make go.modules.tidy.check 
18:11:42 [ .. ] verify go modules dependencies are tidy
18:11:43 [ OK ] go modules are tidy
```
Is there something I'm missing? Thanks in advance.

**Comment by turkenf** [comment]
Could you please rebase to the main and run make generate again?

**Comment by nalbury** [comment]
rebased, reran `make generate`, confirmed tests passed, and pushed the changes

**Comment by jeanduplessis** [comment]
/test-examples="examples/kafka/serverlesscluster.yaml"

**Review by turkenf — APPROVED** [review]
Thank you so much for your contribution @nalbury, LGTM.

**Comment by nalbury** [comment]
Amazing, thank you!

**Pull request merged**